pymc_marketing/mmm/plot.py

```diff
@@ -675,160 +880,210 @@ def saturation_scatterplot(
         dims : dict[str, str | int | list], optional
-            Dimension filters to apply. Example: {"country": ["US", "UK"], "region": "X"}.
+            Dimension filters to apply. Examples:
+            - {"geo": "US"}
+            - {"geo": ["US", "UK"]}
+
             If provided, only the selected slice(s) will be plotted.
-        **plot_kwargs
-            Any other kwargs forwarded to `plot_curve`
-            (for instance `same_axes=True`, `legend=True`, etc.).
+        backend : str, optional
+            Plotting backend to use. Options: "matplotlib", "plotly", "bokeh".
+            If None, uses global config via mmm_plot_config["plot.backend"].
+            Default is "matplotlib".
 
         Returns
         -------
-        fig : plt.Figure
-            Matplotlib figure with your grid.
-        axes : np.ndarray of plt.Axes
-            Array of shape `(n_channels, n_geo)`.
+        PlotCollection
+            arviz_plots PlotCollection object containing the plot.
 
-        """
-        from pymc_marketing.plot import plot_hdi, plot_samples
+            Use ``.show()`` to display or ``.save("filename")`` to save.
 
-        if not hasattr(self.idata, "constant_data"):
-            raise ValueError(
-                "No 'constant_data' found in 'self.idata'. "
-                "Please ensure 'self.idata' contains the constant_data group."
-            )
+        Raises
+        ------
+        ValueError
+            If curve is missing required dimensions ("x" or "channel").
+        ValueError
+            If original_scale=True but channel_contribution_original_scale not in posterior.
 
-        contrib_var = (
-            "channel_contribution_original_scale"
-            if original_scale
-            else "channel_contribution"
-        )
+        See Also
+        --------
+        saturation_scatterplot : Base scatter plot without curves
+        LegacyMMMPlotSuite.saturation_curves : Legacy matplotlib-only implementation
 
-        if original_scale and not hasattr(self.idata.posterior, contrib_var):
-            raise ValueError(
-                f"""No posterior.{contrib_var} data found in 'self.idata'.\n"
-                "Add a original scale deterministic:\n"
-                "    mmm.add_original_scale_contribution_variable(\n"
-                "        var=[\n"
-                "            'channel_contribution',\n"
-                "            ...\n"
-                "        ]\n"
-                "    )\n"
-                """
-            )
-        curve_data = (
-            curve * self.idata.constant_data.target_scale if original_scale else curve
-        )
-        curve_data = curve_data.rename("saturation_curve")
+        Notes
+        -----
+        Breaking changes from legacy implementation:
+
+        - Returns PlotCollection instead of (Figure, Axes)
+        - Lost colors, subplot_kwargs, rc_params parameters
+        - Different HDI calculation (uses arviz_plots instead of custom)
```

Comment:
looking at the deleted code, it looks like before `az.plot_hdi` was used, so the HDI calculation hasn't really changed. The only difference is internal to this method. Now hdi computation is explicit (but still implicit for users) whereas before both hdi computation and plotting was delegated to arviz in a single call.